app/controllers/user.js

```diff
@@ -37,6 +37,54 @@ exports.newUser = (req, res, next) => {
   }
 };
 
+exports.newAdmin = (req, res, next) => {
+  if (!req.user.isAdmin) {
+    next(errors.invalidUser('User does not have access to this resource.'));
+  }
+
+  const admin = req.body
+    ? {
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        password: req.body.password,
+        email: req.body.email
+      }
+    : {};
+
+  const errMsg = validation.validateUser(admin);
+  if (errMsg.length > 0) {
+    next(errors.invalidUser(errMsg));
+  } else {
+    return bcrypt
```

Comment:
Some duplicated logic with newUser, wrap some logic for createOrUpdate an user (avoid do that in controller)